src/modules/notifSettings/components/NotificationSettingsFooter.tsx

```diff
@@ -0,0 +1,232 @@
+
+  const handleSaveSettings = async () => {
+    let userPushInstance = userPushSDKInstance;
+    if (!userPushInstance.signer) {
+      userPushInstance = await handleConnectWalletAndEnableProfile({ wallet });
+      if (!userPushInstance) {
+        return;
+      }
+    }
+
+    setErrorMessage('');
+    if (newSettings.length > 0) {
+      const newsettingData: ChannelSetting[] = newSettings.map((setting) => {
+        if (setting.type === 1) {
+          return {
+            type: setting.type,
+            description: setting.description,
+            default: setting.default ? 1 : 0,
+          };
+        } else {
+          return {
+            type: setting.type,
+            description: setting.description,
+            default: setting.default,
+            data: {
+              lower: setting.lowerLimit,
+              upper: setting.upperLimit,
+              ticker: setting.ticker,
+              enabled: setting.enabled,
+            },
+          };
+        }
+      });
+
+      createNotificationSettings(
+        {
+          userPushSDKInstance: userPushInstance,
+          settings: newsettingData,
+        },
+        {
+          onSuccess: (response) => {
+            console.log('onSuccess >>>', response);
+            if (response.transactionHash) {
+              console.log('Call channel details refetch and navigate to dashboard page');
+              refetchChannelDetails();
+              navigate(`${APP_PATHS.ChannelDashboard}/${account}`)
+            }
+          },
+          onError: (error: any) => {
+            console.log('Error in adding setting', error);
+            setErrorMessage('Error in saving settings. Please try again later');
+          },
+        }
+      );
+    }
+  };
+
+  const settingsChanged = useMemo(() => {
+    if (!channelSettings) return false; // if there are no channel settings
+    if (newSettings.length !== channelSettings.length) return true; // new settings length is not equal to channel settings
+    let isUnchanged = true;
+    for (let i = 0; i < newSettings.length; i++) {
+      const setting1 = newSettings[i];
+      const setting2 = channelSettings[i];
+      if (setting1.type === 1) {
+        isUnchanged =
+          isUnchanged &&
+          setting1.type === setting2.type &&
+          setting1.description === setting2.description &&
+          setting1.default === setting2.default;
+      } else if (setting1.type === 2) {
+        isUnchanged =
+          isUnchanged &&
+          setting1.type === setting2.type &&
+          setting1.description === setting2.description &&
+          setting1.default === setting2.default &&
+          setting1.enabled === setting2.enabled &&
+          setting1.lowerLimit === setting2.lowerLimit &&
+          setting1.upperLimit === setting2.upperLimit &&
+          setting1.ticker === setting2.ticker;
+      }
+    }
```

Comment:
I think formik offers a builtin way to check if the initial values have changed or not. We can use that.

If not still we can optimise this approach by using forEach instead of for loop.